### DISCUSSION TITLE
How to call site.addsitedir() at runtime?

### DISCUSSION THREAD

**Discussion by jamesbcd** [pyinstaller/pyinstaller#6823] [Help] [answered]
My application uses external python plugins, each of which has its own `site-packages` directory for the plugin's dependencies. When unfrozen, it calls `site.addsitedir()` to made these dependencies available to plugin.

Unfortunately, when frozen, pyinstaller seems to use its own version of the `site` module which doesn't have an `addsitedir()` function. Is there a workaround available for calling `site.addsitedir()` at runtime?

**Comment by rokm** [comment]
PyInstaller-frozen applications are isolated by design, which is why `site` is disabled, as if regular python was started with `python -S`. As you noted, the `site` module is currently also replaced with custom module, but that will go away as part of changes related to [python 3.11 support](https://github.com/pyinstaller/pyinstaller/pull/6783). 

So if you really need `site`, you could try applying the relevant changes from that PR, and see if `site.addsitedir` works in spite of `site` itself being disabled. (Or equivalently, you could probably also test this by trying to run your original code via `python -S` instead of `python`...). 

You could also just add those paths to `sys.path` yourself, unless you really need to handle .pth files in `site-packages`...

**Comment by jamesbcd** [comment, marked as answer]
Okay, that's really helpful, thanks. For now, just appending to sys.path should handle the majority of cases.
